src/Microsoft.ML.Tokenizers/PreTokenizer/PreTokenizer.cs

```diff
@@ -54,11 +64,64 @@ public Split(string token, (int Index, int End) offset)
     /// </summary>
     public abstract class PreTokenizer
     {
+        internal static readonly IReadOnlyList<Split> EmptyList = new List<Split>();
+
         /// <summary>
         /// Splits the given string in multiple substrings at the word boundary, keeping track of the offsets of said substrings from the original string.
         /// </summary>
         /// <param name="sentence">The string to split into tokens.</param>
+        /// <param name="skipSpecialTokens">Indicates whether to skip the special tokens.</param>
         /// <returns>The list of the splits containing the tokens and the token's offsets to the original string.</returns>
-        public abstract IReadOnlyList<Split> PreTokenize(string sentence);
+        public abstract IEnumerable<Split> PreTokenize(string sentence, bool skipSpecialTokens = false);
+    }
+
+    internal readonly struct RegexSplitEnumerable : IEnumerable<Split>
+    {
+        private readonly MatchCollection _matches;
+
+        public RegexSplitEnumerable(string sentence, string pattern)
+        {
+            _matches = Regex.Matches(sentence, pattern);
```

Comment:
The pattern here appears to be constant. This should cache the Regex instance, and ideally use the regex source generator when targeting .NET 7+.